docs/DeveloperGuide.md

```diff
@@ -159,6 +159,50 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Add a class feature
+
+#### Implementation
+
+Step 1. The user launches the application.
+
+Step 2. The user executes `add /c cs2103t` command to add a new class with the class name CS2103T. The `add /c` command calls `Parser#AddClassCommandParser` to retrieve the provided class name argument.
+
+Step 3. A new `Class` is created with the specified class name, an empty student list, an empty class note and empty class schedule.
+
+Step 4. The created `Class` is checked if the same class exists in `UniqueClassList`, then if not `Class` is added to the `UniqueClassList`, then update the `Storage` to save the current state of `EduTrack`.
```

Comment:
Good to have the details and talks about the check. Perhaps, if you see fit, can shorten the sentence to : 
"The created `Class` is added to `UniqueClassList` if it does not already exists in `UniqueClassList`. The `Storage` then updates to save the current state of `EduTrack`."

Reply:
Thanks for your suggestion!